infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/receive/LogReplicationSinkManager.java

```diff
@@ -0,0 +1,286 @@
+package org.corfudb.infrastructure.logreplication.receive;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+import org.corfudb.common.util.ObservableValue;
+import org.corfudb.infrastructure.logreplication.DefaultSiteConfig;
+import org.corfudb.infrastructure.logreplication.LogReplicationConfig;
+import org.corfudb.protocols.wireprotocol.logreplication.LogReplicationEntry;
+import org.corfudb.protocols.wireprotocol.logreplication.LogReplicationEntryMetadata;
+import org.corfudb.protocols.wireprotocol.logreplication.MessageType;
+import org.corfudb.runtime.CorfuRuntime;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.List;
+import java.util.Properties;
+import java.util.UUID;
+
+import static org.corfudb.protocols.wireprotocol.logreplication.MessageType.LOG_ENTRY_MESSAGE;
+import static org.corfudb.protocols.wireprotocol.logreplication.MessageType.SNAPSHOT_MESSAGE;
+
+/**
+ * This class represents the Log Replication Manager at the destination.
+ *
+ * It is the entry point for log replication at the receiver.
+ *
+ * */
+@Slf4j
+public class LogReplicationSinkManager implements DataReceiver {
+    private static final String config_file = "/config/corfu/corfu_replication_config.properties";
+
+    public static final int DEFAULT_READER_QUEUE_SIZE = 1;
+    public static final int DEFAULT_RESENT_TIMER = 5000;
+    public static final int DEFAULT_MAX_RETRY = 5;
+
+    private int ackCycleTime = DEFAULT_RESENT_TIMER/ DEFAULT_MAX_RETRY;
+    private int ackCycleCnt = DEFAULT_READER_QUEUE_SIZE;
+    private int bufferSize = DEFAULT_READER_QUEUE_SIZE;
+
+    private CorfuRuntime runtime;
+    private SinkBufferManager sinkBufferManager;
+    private StreamsSnapshotWriter snapshotWriter;
+    private LogEntryWriter logEntryWriter;
+
+    @Getter
+    private PersistedWriterMetadata persistedWriterMetadata;
+    private RxState rxState;
+
+    @Getter
+    @Setter
+    boolean leader;
+
+    // The roleType is active or standby
```

Comment:
Since active or standby is related to the federation terminology, maybe we could name this receiver/sender..